docs/developer/architecture/logs.rst

```diff
@@ -197,6 +197,444 @@ overview page.
+
+Fluentd
+"""""""
+
+Fluentd will be deployed as a Deployment with constraints to schedule it only
+on infra nodes.
+By default we need at least 2 replicas (except on a single node platform) to be
+sure we always have at least one instance of Fluentd running.
+
+Configuration
+~~~~~~~~~~~~~
+
+Fluent Bit
+""""""""""
+
+Fluent Bit needs to be configured to scrape and handle properly journal and
+containers logs by default.
+Its configuration
+
+For containers logs, we want to add the following labels:
+
+ - node: the node it comes from
+ - namespace: the namespace the pod is running in
+ - instance: the name of the pod
+ - container: the name of the container
+
+For journal we want these labels:
+
+ - node: the host it comes from
+ - unit: the name of the unit generating these logs
+
+.. note::
+
+  What if host != node ?
+
+  Do we want more labels (e.g. "app", "environment") ?
+
+  How do we allow Solutions to add filters/parsers (not in the reqs) ?
+
+This configuration will also be customizable by the user to be able to
+add new routes to push the log streams to, which will be handled by the CSC
+mechanisms, using the ``metalk8s-fluent-bit-config`` ConfigMap in the
+``metalk8s-logging`` Namespace.
+
+The ConfigMap will look like the followings:
+
+.. code-block:: yaml
+
+  apiVersion: v1
+  kind: ConfigMap
+  metadata:
+    name: metalk8s-fluent-bit-config
+    namespace: metalk8s-logging
+  data:
+    config.yaml: |-
+      apiVersion: addons.metalk8s.scality.com
+      kind: FluentBitConfig
+      spec:
+        output: |-
+          [OUTPUT]
+            Name            es
+            Match           *
+            Host            localhost
+            Port            9200
+            Logstash_Format On
+            Replace_Dots    On
+            Retry_Limit     False
+
+.. note::
+
+  Fluent Bit configuration is not a YAML file and looks more like an INI one.
+  CSC is not able to take care of these files and even if we add support for
+  this format, it will be hard to implement the merging with defaults mechanism
+  as blocks can not really be identified, because there is no unique name or ID.
+  Since we only need to provide to the user the possiblity to add a new output,
+  it should be sufficient as we don't need to edit what's deployed by MetalK8s.
+
+Loki
+""""
+
+Loki's configuration is stored as a Secret, as for Fluent Bit, we need
+to expose few parameters to the user for customization.
+We will use the CSC mechanism to provide the interface for customization,
+with a ConfigMap ``metalk8s-loki-config`` in ``metalk8s-logging`` Namespace.
+
+A Salt module will be developed to manipulate this object, so the logic
+can be kept in only one place.
+
+This module must provide necessary methods to show or edit the configuration
+in 2 different ways:
+
+   * simple
+   * advanced
+
+The ``simple`` mode will only display and allow to change some specific
+configuration, such as the retention or the number of replicas, and in an as
```

Comment:
I'm still worried about scaling Loki components independently from one another... Would this fit a "simple" mode?